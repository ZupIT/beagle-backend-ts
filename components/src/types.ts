export interface Accessibility {
  accessibility?: {
    accessible?: boolean,
    accessibilityLabel?: string,
    isHeader?: boolean,
  },
}

export interface Theme {
  styleId?: string,
}

type Contextless<T> = Omit<T, 'context'>

export type HasContext = 'with-context' | 'without-context'

export type MaybeContext<T, AcceptsContext extends HasContext> = AcceptsContext extends 'with-context'
  ? T
  : Contextless<T>
