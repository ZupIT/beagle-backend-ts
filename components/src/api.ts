/* This file is only used for exporting the API documentation. We do this so we can have a full documentation without
exporting types that doesn't make sense. */

/**
 * @module @zup-it/beagle-backend-components
 */

export * from './'
export {
  BaseImageProps, ImageType, LocalImageProps, RemoteImageProps, WithLocalImageUrl, LocalWebImage, WithMobileImageId,
  LocalMobileImage,
} from './components/image'
export { NavigationBar, NavigationBarItem, SafeArea } from './components/screen-component'
export { TabBarItem } from './components/tab-bar'
export * from './style/full-styles'
