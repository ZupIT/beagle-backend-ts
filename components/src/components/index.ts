export { Button } from './button'
export { Container } from './container'
export { GridView, ListView } from './dynamic-list'
export { Image } from './image'
export { PullToRefresh } from './pull-to-refresh'
export { Screen } from './screen'
export { ScrollView } from './scroll-view'
export { TabBar } from './tab-bar'
export { TextInput } from './text-input'
export { Text } from './text'
export { Webview } from './webview'
