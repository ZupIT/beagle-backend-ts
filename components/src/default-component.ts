import { Component, coreNamespace, ComponentConstructor } from '@zup-it/ds-schema-core'
import { StyledComponent } from './style/styled'
import { HasContext, MaybeContext } from './types'

export abstract class DefaultComponent<Props, AcceptsContext extends HasContext = 'with-context'>
  extends Component<Props> {
  constructor(args: MaybeContext<ComponentConstructor<Props>, AcceptsContext>) {
    // as any: the type is correct, I don't understand why TS complains
    super(args as any)
  }

  namespace = coreNamespace
}

export abstract class DefaultStyledComponent<Props> extends StyledComponent<Props> {
  namespace = coreNamespace
}
