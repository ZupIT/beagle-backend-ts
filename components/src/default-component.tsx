import { BeagleJSX, FC, coreNamespace } from '@zup-it/beagle-backend-core'

interface DefaultComponentProps {
  name: string,
  properties?: Record<string, any>,
}

export const DefaultComponent: FC<DefaultComponentProps> = ({ children, ...props }) => (
  // @ts-ignore todo: actually implement jsx and check these errors. This Error is shown only when compiling.
  <component {...props} namespace={coreNamespace}>{children}</component>
)
