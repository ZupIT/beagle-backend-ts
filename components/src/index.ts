/**
 * @module @zup-it/beagle-backend-components
 */

export { Color, colors } from './color'
export * from './components'
export * from './fragments/conditional-render'
export { StyledComponent, WithStyle, StyledComponentProps } from './style/styled'
export { SimpleStyle as Style, createStyleMap } from './style/simple-styles'
export { submitForm } from './actions/submit-form'
