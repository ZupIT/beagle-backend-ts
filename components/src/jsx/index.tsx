import { BeagleJSX as _BeagleJSX, FC } from '@zup-it/beagle-backend-core'
import { Fragment } from '..'

function processChildrenArgs(children: any[]) {
  if (children.length === 0) return undefined
  return children.length === 1 ? children[0] : children
}

export const BeagleJSX = {
  createElement: (jsx: string | FC<any>, props?: Record<string, any>, ...children: any[]): any => {
    const componentChildren = processChildrenArgs(children)
    if (!jsx) return <Fragment>{componentChildren}</Fragment>
    return _BeagleJSX.createElement(jsx, props, componentChildren)
  },
}
