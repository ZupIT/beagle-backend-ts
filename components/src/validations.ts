/**
 * Validates if the argument represents a color, i.e., if it's a string in the formar #RGB, #RGBA, #RRGGBB or #RRGGBBAA.
 *
 * If the argument is not a color, a warning is logged.
 *
 * @param color the value to check.
 */
export function validateColor(color?: any) {
  if (typeof color !== 'string') return
  if (!color.match(/^#([\dA-Fa-f]{3}|[\dA-Fa-f]{4}|[\dA-Fa-f]{6}|[\dA-Fa-f]{8})$/)) {
    throw new Error(
      `Invalid color: ${color}. The expected formats are #RGB, #RGBA, #RRGGBB or #RRGGBBAA, where R, G, B and A are hexadecimal characters representing the amount of red, green, blue and opacity, respectively.`,
    )
  }
}
