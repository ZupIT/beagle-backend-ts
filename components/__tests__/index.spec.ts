import { mapValues } from 'lodash'
import * as components from 'src'

describe('Components module', () => {
  it('should export', () => expect(mapValues(components, v => typeof v)).toMatchSnapshot())
})
