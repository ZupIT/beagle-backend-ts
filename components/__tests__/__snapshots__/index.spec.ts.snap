// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components module should export 1`] = `
Object {
  "Button": "function",
  "Container": "function",
  "Else": "function",
  "GridView": "function",
  "If": "function",
  "Image": "function",
  "LazyComponent": "function",
  "ListView": "function",
  "PageIndicator": "function",
  "PageView": "function",
  "PullToRefresh": "function",
  "ScreenComponent": "function",
  "ScrollView": "function",
  "SimpleForm": "function",
  "StyledComponent": "function",
  "TabBar": "function",
  "Template": "function",
  "Text": "function",
  "TextInput": "function",
  "Then": "function",
  "Touchable": "function",
  "Webview": "function",
  "colors": "object",
  "createStyleMap": "function",
  "submitForm": "function",
}
`;
